src/Resources/contao/classes/FrontendTemplate.php

```diff
@@ -240,103 +239,6 @@ public function hasAuthenticatedBackendUser()
 
 
 	/**
-	 * Add the template output to the cache and add the cache headers
-	 */
-	protected function addToCache()
```

Comment:
I don't really know. But for creating `.mobile` cache files, contao is checking  `$objPage->mobileLayout > 0` but not for getting them from cache: https://github.com/contao/core/blob/master/system/modules/core/controllers/FrontendIndex.php#L392

I never have a mobile layout defined in themes.

So at the end the system never creates cache files for mobile devices - but [checks their existence](https://github.com/contao/core/blob/master/system/modules/core/controllers/FrontendIndex.php#L397).

And with my hook Contao generates `.mobile` cache files anyway.